File: java/src/main/java/de/sven_torben/serialization_benchmark/logging/AbstractDetailsLogger.java
package de.sven_torben.serialization_benchmark.logging;

import java.util.Arrays;

public abstract class AbstractDetailsLogger implements Logger {

	@Override
	public final void log(final String serializerName, final int bytes,
			final long[] writeTimes, final long[] readTimes) {
		log(
				serializerName, 
				bytes, 
				writeTimes.length,
				Arrays.stream(writeTimes).parallel().min().getAsLong(),
				Arrays.stream(readTimes).parallel().min().getAsLong(),
				Arrays.stream(writeTimes).parallel().average().getAsDouble(),
				Arrays.stream(readTimes).parallel().average().getAsDouble(),
				Arrays.stream(writeTimes).parallel().max().getAsLong(),
				Arrays.stream(readTimes).parallel().max().getAsLong()
		);

	}

	protected abstract void log(String serializerName, int bytes, int iterations,
			long minWrite, long minRead, double avgWrite, double avgRead,
			long maxWrite, long maxRead);

}

File: net/Logging/AbstractDetailsLogger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace De.Sven_Torben.Serialization_Benchmark.Logging
{
    abstract class AbstractDetailsLogger : Logger
    {
        public void Log(string serializerName, int bytes, long[] writeTimes, long[] readTimes)
        {
            Log(
                serializerName,
                bytes,
                writeTimes.Length,
                writeTimes.AsParallel().Min(),
                readTimes.AsParallel().Min(),
                writeTimes.AsParallel().Average(),
                readTimes.AsParallel().Average(),
                writeTimes.AsParallel().Max(),
                readTimes.AsParallel().Max()
            );
        }

        protected abstract void Log(String serializerName, int bytes, int iterations, long minWrite, long minRead, double avgWrite, double avgRead, long maxWrite, long maxRead);
    }
}
